src/main/java/ro/developmentfactory/Cinema/PessimisticLocking.java

```diff
@@ -0,0 +1,32 @@
+package ro.developmentfactory.Cinema;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class PessimisticLocking {
+    private final int totalSeats = 100;
+    private int occupiedSeats = 0;
+    private final Lock lock = new ReentrantLock();
+
+    public void bookSeats(int numSeats) {
```

Comment:
please think about return type here and in `OptimisticLocking` - we have a method and its user should know if it succeeded or not, and if not - for which reason. and not from a string that you print! the easier way would be to return a `boolean` with true when booking succeeded and throw different kind of exceptions for different reasons. see how you can create custom exceptions in Java